src/ecs/commands/runCommandInContainer.ts

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as vscode from 'vscode'
+import * as picker from '../../shared/ui/picker'
+import { Window } from '../../shared/vscode/window'
+import { getLogger } from '../../shared/logger'
+import { ChildProcess } from '../../shared/utilities/childProcess'
+import { EcsContainerNode } from '../explorer/ecsContainerNode'
+import { recordEcsRunExecuteCommand } from '../../shared/telemetry/telemetry.gen'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+import { extensionSettingsPrefix } from '../../shared/constants'
+import { getIdeProperties } from '../../shared/extensionUtilities'
+import { showOutputMessage } from '../../shared/utilities/messages'
+import { ext } from '../../shared/extensionGlobals'
+
+export async function runCommandInContainer(
```

Comment:
Can we break this function up? It's doing a lot of things with many early returns.